Src/java/elm-fhir/src/main/java/org/cqframework/cql/elm/requirements/fhir/DataRequirementsProcessor.java

```diff
@@ -0,0 +1,662 @@
+package org.cqframework.cql.elm.requirements.fhir;
+
+import org.cqframework.cql.cql2elm.CqlTranslatorOptions;
+import org.cqframework.cql.cql2elm.LibraryManager;
+import org.cqframework.cql.cql2elm.NamespaceManager;
+import org.cqframework.cql.cql2elm.model.TranslatedLibrary;
+import org.hl7.cql.model.IntervalType;
+import org.hl7.cql.model.ListType;
+import org.hl7.cql.model.NamedType;
+import org.hl7.elm.r1.*;
+import org.hl7.elm.r1.Element;
+import org.hl7.elm.r1.Expression;
+import org.hl7.fhir.r5.model.*;
+import org.hl7.fhir.r5.model.Library;
+import org.hl7.fhir.utilities.validation.ValidationMessage;
+import org.cqframework.cql.elm.requirements.ElmRequirement;
+import org.cqframework.cql.elm.requirements.ElmRequirements;
+import org.cqframework.cql.elm.requirements.ElmRequirementsContext;
+import org.cqframework.cql.elm.requirements.ElmRequirementsVisitor;
+
+import javax.xml.bind.JAXBElement;
+import java.io.Serializable;
+import java.util.*;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DataRequirementsProcessor {
```

Comment:
Right now the structures are the same, so this is fine, but in the very near future they will start to diverge and this will become a bigger issue. At that point, we will need to provide a solution in the publisher, and that solution will govern the direction we take here. Given that the publisher currently does version-mapping, it will likely take that form and we'll have to figure out specific workarounds for lossy conversions.